src/highlevel/bidib_highlevel_setter.c

```diff
@@ -515,19 +565,36 @@ int bidib_emergency_stop_train(const char *train, const char *track_output) {
 
 // Must be called with bidib_state_trains_rwlock >=read acquired and
 // bidib_state_track_rwlock >=read acquired.
+
+/**
+ * Get the current bit(s) for the train peripherals.
+ * Shall only be called with bidib_state_trains_rwlock >= read acquired, 
+ * and with  trackstate_trains_mutex acquired.
+ * 
+ * @param train 
+ * @param start 
+ * @param end 
+ * @param bits (out-parameter)
+ */
 static void bidib_get_current_train_peripheral_bits(const t_bidib_train *const train, size_t start,
                                                     size_t end, uint8_t *bits) {
-	const t_bidib_train_state_intern * train_state = 
+	const t_bidib_train_state_intern *train_state = 
 	                              bidib_state_get_train_state_ref(train->id->str);
 	for (size_t i = 0; i < train->peripherals->len; i++) {
 		const t_bidib_train_peripheral_mapping *const mapping_i = 
 		               &g_array_index(train->peripherals, t_bidib_train_peripheral_mapping, i);
 		if (mapping_i->bit >= start && mapping_i->bit <= end) {
+			///NOTE: j is actually not used in the nested loop, is that intentional?
```

Comment:
Probably copy-paste error

Reply:
yeah, but in what direction: Should j occur somewhere? I dont see where it makes sense, as the train peripherals are working as-is. Or was it just accidentally copying two loops when one would have been enough? I think its probably the latter.

Reply:
If I remember correctly, the other data types contain lists of boards that contain lists of things, but train->peripherals is just a simple list.